src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+        capacity = DEFAULT_CAPACITY;
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value, null);
+        Node<K, V> currentNode = table[getIndex(key)];
+        if (currentNode == null) {
+            size++;
+            resizeIfNeeded();
+            table[getIndex(key)] = newNode;
+            return;
+        }
+        if (Objects.equals(key, currentNode.key)) {
+            newNode.next = currentNode.next;
+            table[getIndex(key)] = newNode;
+            return;
+        }
+        while (currentNode.next != null) {
+            if (Objects.equals(key, currentNode.next.key)) {
+                newNode.next = currentNode.next.next;
+                currentNode.next = newNode;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        size++;
+        if (resizeIfNeeded()) {
+            put(newNode.key, newNode.value);
+        }
+        currentNode.next = newNode;
     }
```

Comment:
In the `put` method, you are using `getIndex(key)` multiple times. It would be better to store the result of `getIndex(key)` in a variable at the start of the method and use this variable instead. This would improve the readability of your code and also slightly improve performance as you are not recalculating the index multiple times. For example:
java
int index = getIndex(key);
Node<K, V> currentNode = table[index];
// ...
table[index] = newNode;
```
Remember to create informative variable and method names. Writing proper variable names can be a highly valuable investment to the quality of your code.